### PR TITLE
HSEARCH-4556 Upgrade to JDK 11 / HSEARCH-4555 Upgrade JBoss logging lib / HSEARCH-4787 Upgrade to Mockito 5 / HSEARCH-4393 Upgrade to Lucene 9.7 / HSEARCH-4845 Use a different field for retrieving a root object id

### PR DESCRIPTION
**Pull request by marko-bekhta**
https://hibernate.atlassian.net/browse/HSEARCH-4556
https://hibernate.atlassian.net/browse/HSEARCH-4555
https://hibernate.atlassian.net/browse/HSEARCH-4787
https://hibernate.atlassian.net/browse/HSEARCH-4393
https://hibernate.atlassian.net/browse/HSEARCH-4845

### CODE REVIEW

File: backend/lucene/src/main/java/org/hibernate/search/backend/lucene/lowlevel/docvalues/impl/TextMultiValues.java [hunk 1]
```diff
@@ -22,6 +22,10 @@
  */
 public abstract class TextMultiValues {
 
+	@SuppressWarnings("deprecation")
+	// TODO: monitor when this is removed in Lucene?
```
Comment (yrodiere) [inline backend/lucene/src/main/java/org/hibernate/search/backend/lucene/lowlevel/docvalues/impl/TextMultiValues.java:26]: TODO to be addressed.

File: backend/lucene/src/main/java/org/hibernate/search/backend/lucene/types/predicate/impl/LuceneExistsPredicate.java [hunk 2]
```diff
@@ -43,41 +42,22 @@ public SearchPredicate build() {
...
-		}
-
-		@Override
-		protected Query buildQuery() {
-			return new NormsFieldExistsQuery( absoluteFieldPath );
-		}
-	}
-
-	public static class DocValuesBasedFactory<F>
+	public static class DocBasedFactory<F>
```
Comment (yrodiere) [inline backend/lucene/src/main/java/org/hibernate/search/backend/lucene/types/predicate/impl/LuceneExistsPredicate.java:45]: Maybe rather `DocValuesOrNormsBasedFactory`?

File: backend/lucene/src/main/java/org/hibernate/search/backend/lucene/types/predicate/impl/LuceneExistsPredicate.java [hunk 3]
```diff
@@ -43,41 +42,22 @@ public SearchPredicate build() {
...
-		public DocValuesBasedBuilder<F> create(LuceneSearchIndexScope<?> scope, LuceneSearchIndexValueFieldContext<F> field) {
-			return new DocValuesBasedBuilder<>( scope, field );
+		public DocBasedBuilder<F> create(LuceneSearchIndexScope<?> scope, LuceneSearchIndexValueFieldContext<F> field) {
+			return new DocBasedBuilder<>( scope, field );
 		}
 	}
 
-	private static class DocValuesBasedBuilder<F> extends AbstractBuilder<F> implements ExistsPredicateBuilder {
-		private DocValuesBasedBuilder(LuceneSearchIndexScope<?> scope, LuceneSearchIndexValueFieldContext<F> field) {
+	private static class DocBasedBuilder<F> extends AbstractBuilder<F> implements ExistsPredicateBuilder {
```
Comment (yrodiere) [inline backend/lucene/src/main/java/org/hibernate/search/backend/lucene/types/predicate/impl/LuceneExistsPredicate.java:53]: Maybe rather `DocValuesOrNormsBasedBuilder`?

File: documentation/src/main/asciidoc/migration/index.adoc [hunk 4]
```diff
@@ -43,9 +43,13 @@ Hibernate Search's requirements did not change in version {hibernateSearchVersio
 [[data-format]]
 == Data format and schema changes
```
Comment (yrodiere) [inline documentation/src/main/asciidoc/migration/index.adoc:44]: Maybe also add something to the "Requirements" section, about JDK 11 and Lucene 9.

File: integrationtest/java/modules/orm-coordination-outbox-polling-elasticsearch/pom.xml [hunk 5]
```diff
@@ -44,6 +44,12 @@
             <artifactId>hibernate-search-mapper-orm-coordination-outbox-polling</artifactId>
         </dependency>
 
+        <!-- Only needed here so we can require it in module info for Avro -->
+        <dependency>
+            <groupId>org.apache.logging.log4j</groupId>
+            <artifactId>log4j-core</artifactId>
+        </dependency>
```
Comment (yrodiere) [inline integrationtest/java/modules/orm-coordination-outbox-polling-elasticsearch/pom.xml:51]: Shouldn't this be just log4j-api?

Also... would this work if the dependency is marked as `<scope>provided</scope>`? I don't want to have to align the dependency on Avro...

File: jakarta/integrationtest/java/modules/orm-coordination-outbox-polling-elasticsearch/pom.xml [hunk 6]
```diff
@@ -46,6 +46,12 @@
             <artifactId>hibernate-search-mapper-orm-coordination-outbox-polling-jakarta</artifactId>
         </dependency>
 
+        <!-- Only needed here so we can require it in module info for Avro -->
+        <dependency>
+            <groupId>org.apache.logging.log4j</groupId>
+            <artifactId>log4j-core</artifactId>
```
Comment (yrodiere) [inline jakarta/integrationtest/java/modules/orm-coordination-outbox-polling-elasticsearch/pom.xml:52]: Same comment as elsewhere, shouldn't this be `log4j-api`, and can we reduce the scope?

File: orm6/integrationtest/java/modules/orm-coordination-outbox-polling-elasticsearch/pom.xml [hunk 7]
```diff
@@ -54,6 +54,12 @@
             <version>${version.net.bytebuddy}</version>
         </dependency>
 
+        <!-- Only needed here so we can require it in module info for Avro -->
+        <dependency>
+            <groupId>org.apache.logging.log4j</groupId>
+            <artifactId>log4j-core</artifactId>
```
Comment (yrodiere) [inline orm6/integrationtest/java/modules/orm-coordination-outbox-polling-elasticsearch/pom.xml:60]: Same comment as elsewhere, shouldn't this be `log4j-api`, and can we reduce the scope?

**Review by yrodiere — COMMENTED** [review]
Thanks! A few comments below.

File: pom.xml
[the remark this answers is not in the document]
Comment (marko-bekhta) [reply pom.xml:1]: > I believe you can also upgrade version.processor.plugin to something that does not mention jdk8?

Good point 😃 I think what misled me was the plugin description (I've assumed that non jdk8 would be for jdk6 😨):

>A maven plugin to process annotation for jdk6 at compile time

but looking at the Github: 4.0+ - Release based on JDK9 and above. Will change

File: integrationtest/java/modules/orm-coordination-outbox-polling-elasticsearch/pom.xml (hunk 5 above)
Comment (marko-bekhta) [reply integrationtest/java/modules/orm-coordination-outbox-polling-elasticsearch/pom.xml:51]: I've tried it locally, and looks like it is ok to build/run with provided scope.

**Review by yrodiere — APPROVED** [review]
LGTM, thanks!

**Pull request merged**